src/main/scala/org/apache/spark/mllib/linalg/distributed/BetterBlockMatrix.scala

```diff
@@ -0,0 +1,188 @@
+package org.apache.spark.mllib.linalg.distributed
```

Comment:
I don't think I use sneaky grid partitioner in this file. The only place I need the sneaky partitioner is in `BlockMatrixIsDistributedMatrix` from the previous PR. I need to enforce partitioning when generating a new `BlockMatrix`. John C also used it to create an efficient `IndexedRowMatrix` to dense `BlockMatrix`.